src/services/AudioMixer.cxx

```
@@ -185,28 +312,47 @@ void AudioMixer::playSoundtrack(SoundtrackUPtr& track)
   if(!track)
     throw RuntimeError("Received an invalid soundtrack");
   
+  #ifndef USE_OPENAL_SOFT
   track->Channel = Mix_PlayChannel(track->Channel.get(), track->Chunks, 0);
   if(track->Channel.get() == -1)
   {
-    /* We might need to allocate more channels */
-    int num_alloc = Mix_AllocateChannels(-1);
-    Mix_AllocateChannels(2 * num_alloc);
-    track->Channel = Mix_PlayChannel(track->Channel.get(), track->Chunks, 0);
-    if(track->Channel.get() == -1)
-      throw RuntimeError("Failed to play track " + track->ID.get() + ": " + SDL_GetError());
+	/* We might need to allocate more channels */
+	int num_alloc = Mix_AllocateChannels(-1);
+	Mix_AllocateChannels(2 * num_alloc);
+	track->Channel = Mix_PlayChannel(track->Channel.get(), track->Chunks, 0);
+	if(track->Channel.get() == -1)
+	  throw RuntimeError("Failed to play track " + track->ID.get() + ": " + SDL_GetError());
   }
   m_Playing.push_front(&track);
   track->isPlaying = true;
+
+  #else
+  std::cout << "OpenAL Soft called to play track! \n";
+  if(track->source != 0)
+  {
+	  alSourcePlay(track->source);
+	  m_Playing.push_front(&track);
+	  track->isPlaying = true;
+  }
+  else
+  {
+	  std::cout << "Unable to play track because it's source is uninitialized! \n";
```

I would throw an exception here
```suggestion
	  throw RuntimeError{"Unable to play track because it's source is uninitialized"};
```

---

Resolved. Throwing an exception.